### ISSUE TITLE
TypeError: Cannot read property 'clone' of undefined

### ISSUE THREAD

**Issue by ganesh-papola** [BugiDev/react-native-calendar-strip#257]
Android/ios release error in date clone
```
TypeError: Cannot read property 'clone' of undefined

This error is located at:
    in CalendarStrip (at DatePicker.js:64)
    in RCTView (at View.js:34)
    in View (at DatePicker.js:55)
    in App (at EventList.js:259)
    in RCTView (at View.js:34)
    ...
```

**Comment by peacechen** [comment]
This project doesn't have a file named `DatePicker.js`. Is that in your code? As with the other issue that you posted, please include a reproducible demo or code snippet. Use the example project in this repo as a base if that's convenient.

**Comment by ganesh-papola** [comment]
```
          <CalendarStrip
            scrollable
            minDate={new Date()}
            // calendarAnimation={{type: 'parallel', duration:800, paddingTop: 10, paddingBottom: 10}}
            // daySelectionAnimation={{ type: 'background', duration: 300, barderRadius: 1, }}
            style={{ height: hp(100), paddingTop: hp(20), paddingBottom: hp(20) }}
            calendarHeaderStyle={{ color: 'white', paddingBottom: hp(7) }}
            calendarColor={COLORS.PRIMARY}
            dateNumberStyle={{ color: 'white', }}
            dateNameStyle={{ color: 'white', }}
            iconContainer={{ flex: 0.1 }}
            // locale={{ name: 'fr', config: {} }}
            selectedDate={this.state.selectedDate}
            // markedDates={this.state.markedDates}
            onDateSelected={this.onDateSelected}
            onWeekChanged={this.props.onWeekChanged?this.props.onWeekChanged:null}
            useIsoWeekday={false}
            iconLeft={require('../images/left-arrow.png')}
            iconRight={require('../images/right-arrow.png')}
            highlightDateNameStyle={'black'}
            highlightDateNumberStyle={'black'}
            calendarAnimation={{ type: 'sequence', duration: 30 }}
            daySelectionAnimation={{ type: 'border', duration: 200, borderWidth: 1, borderHighlightColor: 'white', highlightColor: 'black' }}
          />
          
          
          ```
          this is code in DatePicker.js
          
          
          
<img width="1075" alt="Screenshot 2021-01-10 at 7 14 39 PM" src="https://user-images.githubusercontent.com/19851598/104124677-19440080-5378-11eb-897a-00036dcf0929.png">

**Comment by peacechen** [comment]
That's not enough information. What are the state variable values passed to CalendarStrip? Either set breakpoints in CalendarStrip to determine which line the error is from, or create a Snack that reproduces the bug.

**Comment by hans-permana** [comment]
I encountered the same issue. I use the version **2.1.4**

This is fixed when the startingDate is given a value. Maybe this should be made mandatory in the TS? Also, I found it tricky dealing with momentjs object since this object is mutable. I understand now why there are a lot of clones in the code. Anyway, awesome library!

**Comment by peacechen** [comment]
Thanks @hans-permana for the additional info. The `startingDate` prop is checked for null and set to today's date. Which line is the clone exception happening?

I agree that Moment can be unintuitive. There's a project called date-io that abstracts numerous date libraries. Please consider contributing to this project.

**Comment by peacechen** [comment]
I debugged this in the example project by supplying a `selectedDate` prop set to this.state.selectedDate initially undefined. When a date is selected, the `onDateChange` callback updates state and trickles back into CalendarStrip.  However the prop update tried to clone `startingDate` but that prop is undefined. I've added a guard which will be published in 2.1.5.
